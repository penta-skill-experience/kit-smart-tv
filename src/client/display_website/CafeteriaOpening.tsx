import * as React from "react";
import axios from "axios";

export class CafeteriaOpening extends React.Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            data: ""
        };
    }

    componentDidMount() {
        this.getCafeteriaOpening();
        setInterval(() => this.getCafeteriaOpening(), 1000000);
    }

    async getCafeteriaOpening() {

        await axios.get('')
            .then(resp => {
                console.log(resp.data);
                this.setState({
                });
            })
            .catch(function (error) {
                console.log(error);
            });
    }

    render() {
        return <div className="grid grid-col-3 grid-flow-col gap-3 box-border">
        </div>;
    }
}