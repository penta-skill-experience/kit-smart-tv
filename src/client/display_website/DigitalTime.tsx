import * as React from "react";
const weekday = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"];
export class DigitalTime extends React.Component<any, any> {
    private intervalID: NodeJS.Timer;
    constructor(props) {
        super(props);
        this.state = {
            time: new Date().toLocaleTimeString(navigator.language, {hour: '2-digit', minute:'2-digit'})
                .replace(/(:\d{2}| [AP]M)$/, ""),
            date: new Date().toLocaleDateString(navigator.language, {day: 'numeric', month:'long'}),
            day: new Date().getDay()
        };
    }
    componentDidMount() {
        this.intervalID = setInterval(
            () => this.tick(),
            1000
        );
    }
    componentWillUnmount() {
        clearInterval(this.intervalID);
    }
    tick() {
        this.setState({
            time: new Date().toLocaleTimeString(navigator.language, {hour: '2-digit', minute:'2-digit'}),
            date: new Date().toLocaleDateString(navigator.language, {day: 'numeric', month:'long'}),
            day: new Date().getDay()


        });
    }
    render() {
        return <div className="flex-col">
            <div className="font-light text-white sm:text-2xl md:text-4xl lg:text-6xl xl:text-7xl 2xl:text-8xl 4xl:text-9xl 8xl:text-10xl sm:text-left">{this.state.time}</div
            ><div className="text-white sm:text-xs md:text-sm lg:text-base lg:text-xl 2xl:text-2xl 4xl:text-4xl 8xl:text-5xl sm:text-left sm:-mt-1 md:-mt-2 lg:-mt-3 xl:-mt-5 2xl:-mt-6 4xl:-mt-8 8xl:-mt-10">{weekday[this.state.day]}, {this.state.date}</div>
        </div>
    }
}