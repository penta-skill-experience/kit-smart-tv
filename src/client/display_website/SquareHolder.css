#squareHeld {
    backdrop-filter: blur(25px);
}