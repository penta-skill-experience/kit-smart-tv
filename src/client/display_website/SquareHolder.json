{
  "SCROLL_REFRESH": 1000,
  "RATIO": 1.6,
  "ERROR_CLEAR_INTERVAL": 30000
}
