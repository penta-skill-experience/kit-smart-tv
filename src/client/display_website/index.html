<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Smart TV</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
