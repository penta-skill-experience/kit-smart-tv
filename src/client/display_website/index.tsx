import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {RootComponent} from "./RootComponent";

ReactDOM.render(
    <RootComponent/>,
    document.getElementById("root")
);