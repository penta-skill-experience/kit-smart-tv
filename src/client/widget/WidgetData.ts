export class WidgetData {

    widgetId: string;
    location: number;
    fill: boolean;
    rawConfig: Object;

    constructor(widgetId: string, location: number, fill: boolean, rawConfig: Object) {
        this.widgetId = widgetId;
        this.location = location;
        this.fill = fill;
        this.rawConfig = rawConfig;
    }
}