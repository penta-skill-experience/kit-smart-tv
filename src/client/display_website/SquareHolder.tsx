import * as React from "react";
import * as SquareHolderConfig from "./SquareHolder.json";
import "./SquareHolder.css";

interface SquareHolderProps {
    title: string;
    titleColor: string;
    accentColor: string;
    specialBoldFontColor: string;
    specialSubtleFontColor: string;
}

export class SquareHolder extends React.Component<SquareHolderProps, any> {

    pageScroll = function () {
        if (this.state.uniqueID === null || document.getElementById(this.state.uniqueID) === null) {
            return;
        }
        document.getElementById(this.state.uniqueID).scrollBy(0, this.state.scroll); // horizontal and vertical scroll increments
        if ((document.getElementById(this.state.uniqueID).scrollTop +
                document.getElementById(this.state.uniqueID).clientHeight) >=
            document.getElementById(this.state.uniqueID).scrollHeight || document.getElementById(this.state.uniqueID).scrollTop == 0) {
            this.state.scroll = -this.state.scroll;
        }
    }
    doesOverflow = function () {
        return document.getElementById(this.state.uniqueID).scrollHeight >= document.getElementById(this.state.uniqueID).clientHeight;
    }
    randomID = function () { //generate random ID to make scrolling behavior unique for each Squareholder
        let part = function () {
            return (Math.random().toString(10).substring(2));
        }
        return ("id" + part() + part() + part());
    };

    constructor(props) {
        super(props);
        this.state = {
            uniqueID: this.randomID(),
            hasError: false,
            error: undefined,
            scroll: SquareHolderConfig.SCROLL_SPEED
        };
        setInterval(() => this.pageScroll(), SquareHolderConfig.SCROLL_REFRESH);
        //(this.doesOverflow() ? setInterval(() => this.pageScroll(), 50):"");
    }

    static getDerivedStateFromError(error) {
        return {
            hasError: true,
            error: error,
        };
    }

    render() {
        return <div className = "box-border" style={{
            height: "45vh",
            width: "31vw",
            boxSizing: "border-box"
        }}>
            <div className={"w-full h-full shadow-2xl rounded-2xl"} id="squareHeld" style={{
                backgroundColor: this.props.accentColor,
                boxSizing: "border-box",
                overflow: "hidden"
            }}>
                <div
                    className={"sm:px-2 lg:px-3 xl:px-4 4xl:px-5 8xl:px-6 font-light leading-normal sm:text-base md:text-xl lg:text-2xl xl:text-4xl 2xl:text-5xl 8xl:text-6xl"}
                    style={{
                        color: this.props.titleColor
                    }}>
                    {this.props.title}
                </div>
                <div
                    className={"w-full h-full sm:pl-5 sm:pt-1 sm:pb-2 sm:pr-2 xl:pl-8 xl:pr-5 xl:pb-4 4xl:pl-12 overflow-x-scroll scrollbar-hide scroll-smooth overflow-hidden"}
                    id={this.state.uniqueID} style={{
                    scrollBehavior: "smooth"
                }}>
                    {this.state.hasError ?
                        // todo: make design for error message nicer
                        <p>Error while rendering widget: {this.state.error.message}</p>
                        : this.props.children  // only show children if they didn't produce an error
                    }
                </div>
            </div>
        </div>;
    }
}