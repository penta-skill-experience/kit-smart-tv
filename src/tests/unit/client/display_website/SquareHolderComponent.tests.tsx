/**
 * @jest-environment jsdom
 */
import React from "react";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {configure, shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {WidgetData} from "../../../../client/widget/WidgetData";
import {SquareHolder} from "../../../../client/display_website/SquareHolder";
import {
    CafeteriaOpeningDisplayComponent
} from "../../../../client/widget_catalog/cafeteria_opening/CafeteriaOpeningDisplayComponent";

const widgetOne = new WidgetData("cafeteria-menu", 2, {});

//mocking display website persistence to return test values for these tests
//no prototype for static methods
const mathMock = jest.spyOn(Math, "random");

configure({adapter: new Adapter()});

describe("Square Holder Snapshots", () => {

    test("Square holder Snapshots with design values and widget info", async () => {

        mathMock.mockImplementation(() => {
            return 0.5;
        });
        const wrapper = shallow(<SquareHolder displayComponentClass={CafeteriaOpeningDisplayComponent}
                                              rawConfig={""}
                                              title={widgetOne.widgetId}
                                              accentColor="black"
                                              titleColor="black"
                                              specialBoldFontColor="black"
                                              specialSubtleFontColor="black">
        </SquareHolder>);

        jest.useFakeTimers("legacy");
        await new Promise(process.nextTick);
        jest.runAllTimers();
        await new Promise(process.nextTick);
        expect(toJson(wrapper)).toMatchSnapshot();

    });

    afterEach(() => {
        jest.useRealTimers();
        jest.restoreAllMocks();
    });
});