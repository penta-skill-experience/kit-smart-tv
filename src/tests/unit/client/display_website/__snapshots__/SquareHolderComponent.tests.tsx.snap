// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Holder Snapshots Square holder Snapshots with error 1`] = `
<div
  className="xl:py-1 8xl:py-2"
  style={
    Object {
      "height": "45.5vh",
      "width": "30.5vw",
    }
  }
>
  <div
    className="w-full h-full rounded-2xl"
    id="squareHeld"
    style={
      Object {
        "backgroundColor": "black",
        "overflow": "hidden",
      }
    }
  >
    <div
      className="z-20 sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4 font-light leading-normal sm:text-base md:text-2xl lg:text-3xl xl:text-4xl 2xl:text-5xl 8xl:text-7xl"
      style={
        Object {
          "color": "black",
        }
      }
    >
      cafeteria-menu
    </div>
    <div
      className="z-10 sm:pl-5 sm:pb-2 sm:pr-2 xl:pl-8 xl:pr-5 xl:pb-4 4xl:pl-12 scrollbar-hide"
      id="id555"
      style={
        Object {
          "height": "35vh",
          "overflow": "scroll",
        }
      }
    >
      <div
        id="id555"
      >
        <div
          className="w-full h-full font-light leading-normal sm:text-xs lg:text-sm xl:text-base 2xl:text-lg 4xl:text-xl text-center"
        >
          <div
            className="xs:w-10 sm:w-12 md:w-14 base:w-18 lg:w-24 xl:w-35 2xl:w-43 4xl:w-50 8xl:w-60"
          >
            <div />
            <img
              alt="Error_Robot"
              src="https://upload.wikimedia.org/wikipedia/commons/2/24/094-robot-face-3.svg"
            />
            <div />
          </div>
          <span
            className="text-left"
          >
            Render error: 
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Square Holder Snapshots Square holder Snapshots with large overflow 1`] = `
<div
  className="xl:py-1 8xl:py-2"
  style={
    Object {
      "height": "45.5vh",
      "width": "30.5vw",
    }
  }
>
  <div
    className="w-full h-full rounded-2xl"
    id="squareHeld"
    style={
      Object {
        "backgroundColor": "black",
        "overflow": "hidden",
      }
    }
  >
    <div
      className="z-20 sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4 font-light leading-normal sm:text-base md:text-2xl lg:text-3xl xl:text-4xl 2xl:text-5xl 8xl:text-7xl"
      style={
        Object {
          "color": "black",
        }
      }
    >
      cafeteria-menu
    </div>
    <div
      className="z-10 sm:pl-5 sm:pb-2 sm:pr-2 xl:pl-8 xl:pr-5 xl:pb-4 4xl:pl-12 scrollbar-hide"
      id="id555"
      style={
        Object {
          "height": "35vh",
          "overflow": "scroll",
        }
      }
    >
      <div
        id="id555"
      >
        <CafeteriaOpeningDisplayComponent
          config=""
          error={[Function]}
          specialBoldFontColor="black"
          specialSubtleFontColor="black"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Square Holder Snapshots Square holder Snapshots with no element 1`] = `
<div
  className="xl:py-1 8xl:py-2"
  style={
    Object {
      "height": "45.5vh",
      "width": "30.5vw",
    }
  }
>
  <div
    className="w-full h-full rounded-2xl"
    id="squareHeld"
    style={
      Object {
        "backgroundColor": "black",
        "overflow": "hidden",
      }
    }
  >
    <div
      className="z-20 sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4 font-light leading-normal sm:text-base md:text-2xl lg:text-3xl xl:text-4xl 2xl:text-5xl 8xl:text-7xl"
      style={
        Object {
          "color": "black",
        }
      }
    >
      cafeteria-menu
    </div>
    <div
      className="z-10 sm:pl-5 sm:pb-2 sm:pr-2 xl:pl-8 xl:pr-5 xl:pb-4 4xl:pl-12 scrollbar-hide"
      id="id555"
      style={
        Object {
          "height": "35vh",
          "overflow": "scroll",
        }
      }
    >
      <div
        id="id555"
      >
        <CafeteriaOpeningDisplayComponent
          config=""
          error={[Function]}
          specialBoldFontColor="black"
          specialSubtleFontColor="black"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Square Holder Snapshots Square holder Snapshots with small overflow 1`] = `
<div
  className="xl:py-1 8xl:py-2"
  style={
    Object {
      "height": "45.5vh",
      "width": "30.5vw",
    }
  }
>
  <div
    className="w-full h-full rounded-2xl"
    id="squareHeld"
    style={
      Object {
        "backgroundColor": "black",
        "overflow": "hidden",
      }
    }
  >
    <div
      className="z-20 sm:pb-1 lg:pb-2 xl:pb-3 4xl:pb-4 font-light leading-normal sm:text-base md:text-2xl lg:text-3xl xl:text-4xl 2xl:text-5xl 8xl:text-7xl"
      style={
        Object {
          "color": "black",
        }
      }
    >
      cafeteria-menu
    </div>
    <div
      className="z-10 sm:pl-5 sm:pb-2 sm:pr-2 xl:pl-8 xl:pr-5 xl:pb-4 4xl:pl-12 scrollbar-hide"
      id="id555"
      style={
        Object {
          "height": "35vh",
          "overflow": "scroll",
        }
      }
    >
      <div
        id="id555"
      >
        <CafeteriaOpeningDisplayComponent
          config=""
          error={[Function]}
          specialBoldFontColor="black"
          specialSubtleFontColor="black"
        />
      </div>
    </div>
  </div>
</div>
`;
